DataBiosphere/leonardo
# [IA-4788] New deleteAllResources API

__Jira ticket__: https://broadworkbench.atlassian.net/browse/IA-4788
__Design doc__: https://broadworkbench.atlassian.net/wiki/spaces/IA/pages/2972942387/deleteWorkspaces+deleteRuntimes+deleteApps+Design




## Summary of changes



### What

- Implements two new Routes to manage the deletion of all Google Resources:
- - DeleteAllResources -> Delete everything, including sending deletion messages to google cloud (intended to be used by AOU to mitigate cost)
- - CleanupAllResources -> Only perform the clean up in SAM, and the leo databases, but no pubsub messages sent to the cloud (intended to be called by RAWLS to clean up after the google project deletion)

### Why

- This will allow AOU to mitigate cost by deleting resources within a workspace, and also cleanup child resources records in leo and sam once rawls has deleted the underlying google project

## Testing these changes

### What to test

TODO - testing on my BEE
- Create a new google workspace
- Create a couple of runtimes and apps with their attached persistent disks via swagger
- Create an orphaned persistent disk
- Call the deleteAllResources endpoint with `deleteInCloud` set to True and `deleteDisk` set to false
- - Check that the VMs and GKE resources have been deleted
- - Check that all of the persistent disks are still present in the project
- Call the deleteAllResources endpoint with `deleteInCloud` set to True and `deleteDisk` set to true
- - Check that all of the persistent disks are now deleted

- Create the same set of resources as above
- Go into the google console and delete the google project
- - Call the deleteAllResources endpoint with `deleteInCloud` set to False and `deleteDisk` set to True
- - Check that all resources are marked as deleted in the leo and sam db
- - Check that the workspace can be deleted from the UI

### Who tested and where

  - _NB: Rerun automation tests on this PR by commenting `jenkins retest` or `jenkins multi-test`._
